server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -163,8 +163,31 @@ public Task createTask(long id, String type, String action, TaskId parentTaskId,
                 return executor.createTask(id, type, action, parentTaskId, taskInProgress, headers);
             }
         };
-        AllocatedPersistentTask task = (AllocatedPersistentTask) taskManager.register("persistent", taskInProgress.getTaskName() + "[c]",
-                request);
+
+        AllocatedPersistentTask task;
+        try {
+            task = (AllocatedPersistentTask) taskManager.register("persistent", taskInProgress.getTaskName() + "[c]", request);
+        } catch (Exception e) {
+            logger.error("Fatal error registering persistent task [{}] with id [{}] and allocation id [{}], removing from persistent tasks",
+                taskInProgress.getTaskName(), taskInProgress.getId(), taskInProgress.getAllocationId());
+
+            persistentTasksService.sendCompletionRequest(taskInProgress.getId(), taskInProgress.getAllocationId(), e,
```

Comment:
I wonder if it makes sense to move this into a separate method  for clarity.

Reply:
:+1: